crates/meroctl/src/cli/context/list.rs

```diff
@@ -23,13 +24,22 @@ impl ListCommand {
 
         let url = multiaddr_to_url(multiaddr, "admin-api/dev/contexts")?;
         let client = Client::new();
-        let response = client.get(url).send().await?;
+        let keypair = config.identity;
+
+        let timestamp = Utc::now().timestamp().to_string();
```

Comment:
We are creating auth headers and then later checking if they are correct. What is the purpose of that?

Reply:
Not sure if I understand the question.

The auth headers (timestamp and signature) are created on the client side (through the cli) using the node's private key. The server then verifies these headers using the node's public key. That means someone (a bad actor) can't just `curl` our endpoints without access to our nodes private key. The server is not creating the headers, but verifying them.

I might have misunderstood something so open to suggestions.